### PR TITLE
Add Auth0 token interceptor and login changes (closed)

### PR DESCRIPTION
**Pull request by gmarupilla**
I'll resolve the issues this evening.

### CODE REVIEW

File: CRBM-Viz/src/app/Services/auth0.service.ts [hunk 1]
```diff
@@ -22,8 +22,8 @@ export class AuthService {
   redirect = this.returnTo + 'callback';
   auth0Client$ = (from(
     createAuth0Client({
-      domain: 'crbm.auth0.com',
-      client_id: '0NKMjbZuexkCgfWY3BG9C3808YsdLUrb',
+      domain: 'dev-t2vyo8fj.auth0.com',
+      client_id: '6euIYB0EPMSNjsPEjrNZAMmpiuVCRYNB',
```
Comment (bilalshaikh42) [inline CRBM-Viz/src/app/Services/auth0.service.ts]: I assume this is a test domain, not meant to be merged?

File: CRBM-Viz/src/app/Layout/links/links.component.ts [hunk 2]
```diff
@@ -8,5 +9,13 @@ import { AuthService } from 'src/app/Services/auth0.service';
 export class LinksComponent implements OnInit {
   constructor(public auth: AuthService) {}
 
-  ngOnInit() {}
+  ngOnInit() {
```
Comment (bilalshaikh42) [inline CRBM-Viz/src/app/Layout/links/links.component.ts]: Is this meant to be added to this component? This component is intended to list out the various pages that we have on the site, So I don't think it needs to know anything about the authentication. If we want to show the user profile I think that would be best in another component

File: CRBM-Viz/src/app/Pages/about/about.component.ts [hunk 3]
```diff
@@ -23,7 +24,9 @@ export class AboutComponent implements OnInit {
       Praesent cursus ac lorem quis convallis. Nullam semper turpis dolor, ac mollis erat faucibus vel.'
   );
 
-  constructor() {}
+  constructor(public auth: AuthService) {}
```
Comment (bilalshaikh42) [inline CRBM-Viz/src/app/Pages/about/about.component.ts]: Likewise, not sure if this component should need the auth service

File: CRBM-Viz/src/app/app.module.ts [hunk 4]
```diff
@@ -59,6 +64,12 @@ import { ProfileComponent } from './Components/profile/profile.component';
     LinksComponent,
```
Comment (bilalshaikh42) [inline CRBM-Viz/src/app/app.module.ts]: These have already been added to master I believe so a merge may be needed to work out the changes

File: CRBM-Viz/src/app/Layout/links/links.component.ts (hunk 2 above)
Comment (gmarupilla) [reply CRBM-Viz/src/app/Layout/links/links.component.ts]: This is meant to be added to any component which is initialized at very start. This seemed to me a good place to put it. What do you think?

**Comment by gmarupilla** [comment]
I just checked, don't merge it. It has some problems. I will fix them and then raise a fresh PR.
After replying, just close the pull request, don't merge it. Just close it.

**Comment by gmarupilla** [comment]
I just checked, don't merge it. It has some problems. I will fix them and then raise a fresh PR.

**Pull request closed without merging**